src/main/java/com/ss/mqtt/broker/util/TopicUtils.java

```diff
@@ -0,0 +1,99 @@
+package com.ss.mqtt.broker.util;
+
+import static com.ss.rlib.common.util.array.ArrayFactory.newReadOnlyArray;
+import com.ss.mqtt.broker.model.topic.SharedTopicFilter;
+import com.ss.mqtt.broker.model.topic.TopicFilter;
+import com.ss.mqtt.broker.model.topic.TopicName;
+import com.ss.rlib.common.util.array.Array;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.util.StringUtils;
+
+public class TopicUtils {
+
+    private static final TopicFilter INVALID_TOPIC_FILTER = new TopicFilter();
+    private static final TopicName INVALID_TOPIC_NAME = new TopicName();
+    public static final TopicName EMPTY_TOPIC_NAME = new TopicName();
+
+    private static final String SHARE_KEYWORD = "$share";
+    private static final String DELIMITER = "/";
+    public static final String MULTI_LEVEL_WILDCARD = "#";
+    public static final String SINGLE_LEVEL_WILDCARD = "+";
+
+    public static boolean isInvalid(@NotNull TopicFilter topicFilter) {
+        return topicFilter == INVALID_TOPIC_FILTER;
+    }
+
+    public static boolean isInvalid(@NotNull TopicName topicName) {
+        return topicName == INVALID_TOPIC_NAME;
+    }
+
+    public static boolean isShared(@NotNull TopicFilter topicFilter) {
+        return topicFilter instanceof SharedTopicFilter;
+    }
+
+    public static boolean hasWildcard(@NotNull TopicFilter topicFilter) {
+        var topic = topicFilter.getRawTopic();
+        return topic.contains(SINGLE_LEVEL_WILDCARD) || topic.contains(MULTI_LEVEL_WILDCARD);
+    }
+
+    public static @NotNull TopicName buildTopicName(@NotNull String topicName) {
+        if (isInvalidTopicName(topicName)) {
+            return INVALID_TOPIC_NAME;
+        } else {
+            return new TopicName(topicName);
+        }
+    }
+
+    public static @NotNull TopicFilter buildTopicFilter(@NotNull String topicFilter) {
+        if (isInvalidTopicFilter(topicFilter)) {
+            return INVALID_TOPIC_FILTER;
+        } else if (isShared(topicFilter)) {
+            return buildSharedTopicFilter(topicFilter);
+        } else {
+            return new TopicFilter(topicFilter);
+        }
+    }
+
+    public static @NotNull Array<String> splitTopic(@NotNull String topic) {
```

Comment:
do we have any case when we need to have collection of segments? not a native array?